### PR TITLE
Fusion: apply engine labels

### PR DESCRIPTION
**Pull request by jtmcdole**
* Make FusionTester class for easier faking and testing.
* Fetch file list ahead of time - we don't want to fetch these 2x
* Count type of files changed

### CODE REVIEW

**Comment by jtmcdole** [comment]
Fixed the nits from the analyzer. Will continue working on expanding this into other parts of the scheduler.

File: app_dart/lib/src/foundation/utils.dart [hunk 1]
```diff
@@ -37,65 +37,72 @@ Duration twoSecondLinearBackoff(int attempt) {
...
+      }
 
-    log.fine('isFusionRef: fusion ref - ');
-    return true;
-  } catch (e) {
-    log.fine('isFusionRef: not a fusion ref - error: $e');
-    return false;
+      log.fine('isFusionRef: fusion ref - ');
+      return true;
+    } catch (e) {
```
Comment (christopherfujino) [inline app_dart/lib/src/foundation/utils.dart:102]: I worry we will catch errors caused by something other than that it's not a fusion PR here, such as the GitHub API is temporarily down. Maybe we should test that it was specifically 404 http status code?

Comment (jtmcdole) [reply app_dart/lib/src/foundation/utils.dart:102]: Yeah, sounds good.

Comment (jtmcdole) [reply app_dart/lib/src/foundation/utils.dart]: Done.

**Pull request merged**